presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -267,13 +270,65 @@ public void testTimestamp()
         // TODO timestamp is not correctly read (see comment in StandardReadMappings.timestampReadMapping), but testing this is hard because of #7122
     }
 
+    @Test
+    public void testJson()
+    {
+        jsonTestCases(jsonDataType())
+                .execute(getQueryRunner(), postgresCreateAndInsert("tpch.postgresql_test_json"));
+    }
+
+    @Test
+    public void testJsonb()
+    {
+        jsonTestCases(jsonbDataType())
+                .execute(getQueryRunner(), postgresCreateAndInsert("tpch.postgresql_test_jsonb"));
+    }
+
+    private DataTypeTest jsonTestCases(DataType<String> jsonDataType)
+    {
+        return DataTypeTest.create()
+                .addRoundTrip(jsonDataType, "{}")
+                .addRoundTrip(jsonDataType, null)
+                .addRoundTrip(jsonDataType, "null")
+                .addRoundTrip(jsonDataType, "123.4")
+                .addRoundTrip(jsonDataType, "\"abc\"")
+                .addRoundTrip(jsonDataType, "\"text with \\\" quotations and ' apostrophes\"")
+                .addRoundTrip(jsonDataType, "\"\"")
+                .addRoundTrip(jsonDataType, "{\"a\":1,\"b\":2}")
+                .addRoundTrip(jsonDataType, "{\"a\":[1,2,3],\"b\":{\"aa\":11,\"bb\":[{\"a\":1,\"b\":2},{\"a\":0}]}}")
+                .addRoundTrip(jsonDataType, "[]");
+    }
+
+    private static DataType<String> jsonDataType()
+    {
+        return dataType(
+                "json",
+                JSON,
+                value -> "JSON " + formatStringLiteral(value),
```

Comment:
The data format of `json` and `jsonb` could be different right? Can we show that and make sure we test the differences as well?

Reply:
I had added test cases for both json and jsonb data format. If you think we have to add something more, could you please give me the reference?

Reply:
I don't know all the details about the difference between json an jsonb. But i think one of the difference is extra spaces. What I mean here is that the data provided for both json and jsonb are the same. Can we format them so the test data for json are not valid jsonb data, and vice versa?

Reply:
@rongrong Sorry for the delay in this. Currently, Postgres JSON and JSONB data types are mapped to Presto JSON type.